envoy/grpc/async_client_manager.h

```diff
@@ -29,6 +29,10 @@ class AsyncClientManager {
 public:
   virtual ~AsyncClientManager() = default;
 
+  virtual RawAsyncClientSharedPtr
```

Comment:
Add docs please. Is there a way to validate that we've updated existing uses to the cached version? Should we make it more obvious that the non-cached version is not preferred? Or make it fully private?

Reply:
We can make the old create method private and only allow creation of async client through the cached interface.

Reply:
Can we keep the old interface? the cached client introduced some race condition in ext_proc_integration_test. 
stream_ = client_.start(*descriptor, *this, options);
Since client_ is now shared and can outlive the *this reference, there is race condition in ext_proc filter.

Reply:
There might be other filters or utilities that assume exclusive ownership of RawAsyncClient

Reply:
It's probably fine, but can you at least rename this to indicate it's somehow the slow path? E.g. RawUncachedAsyncClient or something.

Reply:
We can leave some comments on both the method. but both the cached client and uncached client are created through the same factory method, are you suggesting an alias or a class wrapper for raw async client?

Reply:
Comments at a minimum, explicit naming for factory methods to make clear the dangers of the uncached version.